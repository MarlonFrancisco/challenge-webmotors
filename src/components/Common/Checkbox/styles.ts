import styled from "styled-components";
import Checkbox from "react-simple-checkbox";

export const CustomCheckbox = styled(Checkbox).attrs((props) => ({
  ...props,
  size: 2,
  tickSize: 3,
  borderThickness: 1,
  color: {
    backgroundColor: "#fff",
    borderColor: props.theme.palette.text.primary,
    uncheckedBorderColor: props.theme.palette.text.primary,
    tickColor: props.theme.palette.primary.main,
  },
}))`
  margin-right: 5px;
  top: 0;
`;

export const Container = styled.div`
  display: flex;
  align-items: center;
  margin-right: 20px;
`;
