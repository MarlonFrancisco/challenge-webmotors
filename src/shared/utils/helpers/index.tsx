import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../styled";

type WithThemeProps = {};

export const WithTheme: React.FC<WithThemeProps> = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);
