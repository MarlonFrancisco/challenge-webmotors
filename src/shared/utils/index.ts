export type Options = {
  withoutPrefix: boolean;
  splitted: boolean;
};

export const formatCurrency = (
  value = 0,
  options?: Options
): string | string[] => {
  const formatted = new Intl.NumberFormat("pt-BR", {
    style: "currency",
    currency: "BRL",
  }).format(value);
  let apllyOptions: string | string[] = formatted;

  if (!options) {
    return formatted;
  }

  if (options.withoutPrefix) {
    apllyOptions = apllyOptions.replace("R$", "").trim();
  }

  if (options.splitted) {
    apllyOptions = apllyOptions.split(",");
  }

  return apllyOptions;
};
