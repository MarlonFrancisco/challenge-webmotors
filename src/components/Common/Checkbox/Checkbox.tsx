import React, { CSSProperties } from "react";
import { Typography } from "../Typography";
import { CustomCheckbox, Container } from "./styles";

export type CheckboxProps = {
  onChange: (isChecked: boolean) => void;
  checked: boolean;
  label: string;
  style?: CSSProperties;
};

const Checkbox: React.FC<CheckboxProps> = ({ label, ...rest }) => {
  return (
    <Container>
      <CustomCheckbox {...rest} />
      <Typography size={10}>{label}</Typography>
    </Container>
  );
};

export default Checkbox;
