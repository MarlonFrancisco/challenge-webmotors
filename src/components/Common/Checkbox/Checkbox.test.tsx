import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Checkbox, { CheckboxProps } from "./Checkbox";
import { WithTheme } from "../../../shared/utils/helpers";

const Wrapper = (props: CheckboxProps) => (
  <WithTheme>
    <Checkbox {...props} />
  </WithTheme>
);

describe("Checkbox", () => {
  let props: CheckboxProps;

  beforeEach(() => {
    props = {
      checked: false,
      label: "Box",
      onChange: jest.fn(),
      style: {},
    };
  });

  it("should render without an error", () => {
    const { container } = render(<Wrapper {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should rneder props", () => {
    const { queryByText } = render(<Wrapper {...props} />);

    expect(queryByText(props.label)).not.toBeNull();
  });

  it("should checked input", () => {
    const { getByRole } = render(<Wrapper {...props} checked />);

    const input = getByRole("checkbox") as HTMLElement;

    expect(input.className).toContain("checked");
  });

  it("should not checked input", () => {
    const { getByRole } = render(<Wrapper {...props} />);

    const input = getByRole("checkbox") as HTMLElement;

    expect(input.className).toContain("unchecked");
  });

  it("should call onChange function", () => {
    const { getByRole } = render(<Wrapper {...props} />);

    const input = getByRole("checkbox") as HTMLElement;

    const triggeredEvent = fireEvent.click(input);

    expect(triggeredEvent).toBeTruthy();
    expect(props.onChange).toHaveBeenCalled();
  });
});
