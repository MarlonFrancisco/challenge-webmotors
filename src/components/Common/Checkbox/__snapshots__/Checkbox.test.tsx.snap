// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render without an error 1`] = `
<div>
  <div
    class="sc-gtssRu dqQBAB"
  >
    <div
      aria-checked="false"
      class="Checkbox sc-bdnylx dGFoQD unchecked"
      role="checkbox"
      tabindex="0"
    >
      <input
        style="display: none;"
        type="checkbox"
      />
      <svg
        height="24px"
        transform="scale(0.75)"
        version="1.1"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          class="square"
          fill="#fff"
          height="20"
          rx="2"
          ry="2"
          stroke="#bababc"
          stroke-width="1px"
          style="transition-delay: 600ms; transition-duration: 180ms;"
          width="20"
          x="2"
          y="2"
        />
        <path
          class="tick"
          d="M6,6 v8 h16"
          fill="none"
          stroke="#ca2430"
          stroke-width="3"
          style="transition-delay: 0ms; transition-duration: 500ms;"
          transform="rotate(-45, 12, 12)"
        />
      </svg>
    </div>
    <h1
      style="font-size: 10px; color: rgb(186, 186, 188); font-weight: 500; text-align: center;"
    >
      Box
    </h1>
  </div>
</div>
`;
